### PR TITLE
Jasmine few gps files

### LINKED ISSUE #115: ValueError when only a few files are read in
Commit hash (the first line of output from `git show`): 9e206cf63bb95a4a52fe208aa059af25341d26c2

**Describe the bug**

For an individual, only 3 GPS files were created over the process of the study. A ValueError occurs when trying to run forest on that individual. It would be nice if this simply generated a warning, or if a more interpretable error code was generated. 

**To reproduce**
download_folder = "user_data"
zf = msync.download(Keyring, "q97eL0qkYiOnPUn0tsQRd1A3", "y7utnxft", ["gps"], time_start = "2000-01-01", time_end = "2022-06-01")
zf.extractall(download_folder)
from forest.constants import Frequency
from forest.jasmine.traj2stats import gps_stats_main

gps_stats_main(
    study_folder = download_folder,
    output_folder = "gps_full_output",
    tz_str = "America/New_York",
    frequency = Frequency.DAILY,
    participant_ids = ["y7utnxft"],
    save_traj = True)

**Observed behavior**
User: y7utnxft
Read in the csv files ...
Collapse data within 10 second intervals ...
Extract flights and pauses ...
Infer unclassified windows ...
Merge consecutive pauses and bridge gaps ...
Selecting basis vectors ...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Volumes/SanDisk/opt/anaconda3/envs/python_3_8/lib/python3.8/site-packages/forest/jasmine/traj2stats.py", line 1018, in gps_stats_main
    imp_table = ImputeGPS(mobmat2, bv_set, parameters.method,
  File "/Volumes/SanDisk/opt/anaconda3/envs/python_3_8/lib/python3.8/site-packages/forest/jasmine/mobmat2traj.py", line 204, in ImputeGPS
    home_x,home_y = locate_home(MobMat,tz_str)
  File "/Volumes/SanDisk/opt/anaconda3/envs/python_3_8/lib/python3.8/site-packages/forest/jasmine/mobmat2traj.py", line 56, in locate_home
    home_index = num_xy.index(max(num_xy))
ValueError: max() arg is an empty sequence

**Expected behavior**
I would expect either an error message or a warning message that says "too few files to run forest"

**Additional context**
This happens because this user only registered for a bit, and they only have 3 GPS data files. 

If nobody else gets to this in the next month or so, I could probably create a fix. The workaround I'm using right now is just manually deleting the folder with only three GPS files. But, it's kind of nice to be able to use the beiwe_id = None argument and have forest run on all users in the folder without a warning happening.

### PR DESCRIPTION
**Pull request by GeorgeEfstathiadis**
* Raise error in inferring home location if GPS trajectories do not contain any nighttime hours
* If error returned from imputing home location proceed to next user in GPS summary

Closes #115

### CODE REVIEW

File: forest/jasmine/traj2stats.py [hunk 1]
```diff
@@ -1035,9 +1035,15 @@ def gps_stats_main(
...
-            imp_table = ImputeGPS(mobmat2, bv_set, parameters.method,
-                                  parameters.switch, parameters.num,
-                                  parameters.linearity, tz_str, pars1)
+            try:
+                imp_table = ImputeGPS(
+                    mobmat2, bv_set, parameters.method,
+                    parameters.switch, parameters.num,
+                    parameters.linearity, tz_str, pars1
+                )
+            except Exception as e:
```
Comment (hackdna) [inline forest/jasmine/traj2stats.py:1044]: `except AssertionError`?

File: forest/jasmine/traj2stats.py [hunk 2]
```diff
@@ -1035,9 +1035,15 @@ def gps_stats_main(
...
+                sys.stdout.write(f"Error: {e}\n")
```
Comment (hackdna) [inline forest/jasmine/traj2stats.py:1045]: `sys.stderr`?

File: forest/jasmine/mobmat2traj.py [hunk 3]
```diff
@@ -44,13 +44,18 @@ def locate_home(MobMat,tz_str):
...
+
     """
     ObsTraj = MobMat[MobMat[:,0]==2,:]
     hours = []
     for i in range(ObsTraj.shape[0]):
         time_list = stamp2datetime((ObsTraj[i,3]+ObsTraj[i,6])/2,tz_str)
         hours.append(time_list[3])
     hours = np.array(hours)
+    assert ((hours>=19)+(hours<=9)).sum()>0, """No home location found: 
+        Too few observations at night"""
```
Comment (hackdna) [inline forest/jasmine/mobmat2traj.py:58]: Any particular reason to choose `assert` vs `RuntimeError` or similar?

**Review by hackdna — CHANGES_REQUESTED** [review]

File: forest/jasmine/mobmat2traj.py [hunk 4]
```diff
@@ -44,13 +44,18 @@ def locate_home(MobMat,tz_str):
...
+        AssertionError: if not enough data to infer home location
+
     """
     ObsTraj = MobMat[MobMat[:,0]==2,:]
     hours = []
     for i in range(ObsTraj.shape[0]):
         time_list = stamp2datetime((ObsTraj[i,3]+ObsTraj[i,6])/2,tz_str)
         hours.append(time_list[3])
     hours = np.array(hours)
+    assert ((hours>=19)+(hours<=9)).sum()>0, """No home location found:
```
Comment (hackdna) [inline forest/jasmine/mobmat2traj.py:57]: Missing whitespace around comparison operators.

File: forest/jasmine/mobmat2traj.py (hunk 3 above)
Comment (GeorgeEfstathiadis) [reply forest/jasmine/mobmat2traj.py:58]: No particular reason, I just thought it was the most suitable error type for this problem. But I can change it to another type if you think there is one more suitable.

**Review by hackdna — APPROVED** [review]
Thank you!

**Pull request merged**